components/FormikZod.tsx

```diff
@@ -0,0 +1,230 @@
+    }
+  }
+
+  // Handle type-specific attributes
+  if (field._def.typeName === 'ZodString') {
+    attributes['type'] = 'text';
+    if (!isNil(field['minLength'])) {
+      attributes['minLength'] = field['minLength'];
+    }
+    if (!isNil(field['maxLength'])) {
+      attributes['maxLength'] = field['maxLength'];
+    }
+  } else if (field._def.typeName === 'ZodNumber') {
+    attributes['type'] = 'number';
+    const minChecks = field._def.checks.filter(check => check.kind === 'min').map(check => check.value);
+    const minValue = max(minChecks); // Get the least restrictive minimum
+    if (!isNil(minValue)) {
+      attributes['min'] = minValue;
+    }
+
+    const maxChecks = field._def.checks.filter(check => check.kind === 'max').map(check => check.value);
+    const maxValue = max(maxChecks); // Get the most restrictive maximum
+    if (!isNil(maxValue)) {
+      attributes['max'] = maxValue;
+    }
+  }
+
+  return attributes;
+};
+
+/**
+ * @returns the Zod schema from a Formik form (if defined)
+ */
+export const getSchemaFromFormik = (formik: FormikValues): AcceptedZodSchema | undefined => {
+  if (formik.status && (formik.status instanceof ZodObject || formik.status instanceof ZodEffects)) {
+    return formik.status;
+  }
+};
+
+/**
+ * A wrapper around `useFormik` that plugs in Zod validation.
+ *
+ * @warning In [their docs](https://formik.org/docs/api/useFormik), Formik discourages using `useFormik` directly. This
+ * component will not work out of the box with `StyledInputFormikField` and other components that rely on Formik context such as `Field` or `FastField`.
+ * If you are trying to access Formik state via context, use `useFormikContext`. Only use this hook if you are NOT using <Formik> or withFormik.
+ */
+export function useFormikZod<Values extends FormikValues = FormikValues>({
+  schema,
+  ...props
+}: Omit<FormikConfig<Values>, 'initialValues' | 'validate' | 'initialValues'> & {
+  schema: AcceptedZodSchema;
+  initialValues: z.infer<typeof schema> & Values;
+}) {
+  const intl = useIntl();
+  const validate = React.useCallback(values => getErrorsObjectFromZodSchema(intl, schema, values), [intl, schema]);
+  return useFormik<Values>({ ...props, validate, initialStatus: schema });
+}
+
+/**
+ * A wrapper around `Formik` that plugs in Zod validation.
+ *
+ * Combined with `StyledInputFormikField` this components provides a simple way to have automatic
+ * form validation and HTML best practices (e.g. setting the `required` and HTML validation attributes).
+ */
+export const FormikZod = ({
+  schema,
+  ...props
+}: Omit<React.ComponentProps<typeof Formik>, 'initialStatus' | 'validate' | 'initialValues'> & {
+  schema: AcceptedZodSchema;
+  initialValues: z.infer<typeof schema>;
+}) => {
+  const intl = useIntl();
+  const validate = React.useCallback(values => getErrorsObjectFromZodSchema(intl, schema, values), [intl, schema]);
+  return <Formik initialStatus={schema} validate={validate} {...props} />;
```

Comment:
I did that in https://github.com/opencollective/opencollective-frontend/pull/9906/files#diff-73d03c9ee3fb986dcb705c27700071aff1c7dc5f6acbfb5fa0ffdd671780cea2R230 where I'm adding a custom global config to the status, but I did not bring it here since it doesn't change anything for the code using the library (it's not supposed to use `status` directly).